theseus/geometry/se2.py

```diff
@@ -133,32 +123,21 @@ def exp_map(tangent_vector: torch.Tensor) -> LieGroup:
 
         cosine, sine = rotation.to_cos_sin()
 
-        sin_theta_by_theta = torch.zeros_like(sine)
-        one_minus_cos_theta_by_theta = torch.zeros_like(sine)
-
-        # Compute above quantities when theta is not near zero
-        idx_regular_thetas = theta.abs() > theseus.constants.EPS
-        if idx_regular_thetas.any():
-            sin_theta_by_theta[idx_regular_thetas] = (
-                sine[idx_regular_thetas] / theta[idx_regular_thetas]
-            )
-            one_minus_cos_theta_by_theta[idx_regular_thetas] = (
-                -cosine[idx_regular_thetas] + 1
-            ) / theta[idx_regular_thetas]
-
-        # Same as above three lines but for small angles
-        idx_small_thetas = theta.abs() < theseus.constants.EPS
-        if idx_small_thetas.any():
-            small_theta = theta[idx_small_thetas]
-            small_theta_sq = small_theta ** 2
-            sin_theta_by_theta[idx_small_thetas] = -small_theta_sq / 6 + 1
-            one_minus_cos_theta_by_theta[idx_small_thetas] = (
-                0.5 * small_theta - small_theta / 24 * small_theta_sq
-            )
+        # Compute the approximations when theta is near to 0
+        small_theta = theta.abs() < SE2.SE2_EPS
+        non_zero = torch.ones(
+            1, dtype=tangent_vector.dtype, device=tangent_vector.device
+        )
+        theta_nz = torch.where(small_theta, non_zero, theta)
+        a = torch.where(
```

Comment:
Maybe add a quick comment that `a ~ (cos - 1) / theta` so that it's easier to read. I prefer this to the `cos_minus_one_by_theta` style that we had before. Similar comment for `b`.

Reply:
Will do. Thanks for the reminder.